objects/src/notes/mod.rs

```diff
@@ -94,13 +94,10 @@ impl Note {
         inputs: &[Felt],
         assets: &[Asset],
         serial_num: Word,
-        sender: AccountId,
-        note_type: NoteType,
-        tag: Felt,
+        metadata: NoteMetadata,
```

Comment:
clippy started complaining about the number of arguments to the constructor. Moved the metadata out.